src/Package/Impl/Packages/R/RPackage.cs

```diff
@@ -140,7 +140,10 @@ internal class RPackage : BasePackage<RLanguageService>, IRPackage {
 
             LogCleanup.Cancel();
             ProjectIconProvider.Close();
-            CsvAppFileIO.Close(Services.FileSystem());
+
+            if (Services?.FileSystem() != null) {
```

Comment:
I think it is better to store reference to `Services?.FileSystem()` inside `Initialize`. `Services` may be disposed here already.